lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1344,11 +1354,25 @@ fini(void)
 
 /* mrs functions */
 
+static void *
+mrs_real_malloc(size_t size)
+{
+	void *p;
+
+	p = REAL(malloc)(size);
+	if (p == NULL)
+		return (p);
+	else if (bound_pointers)
+		return (cheri_setbounds(p, size));
```

Comment:
I think these should be a macro or inline function to reduce explicit instances of `cheri_setbounds`. Philosophically that is because each setbounds is a point potentially requiring audit when you bring in foreign code.

Reply:
Done in the latest push. I also added a comment above the new function to explain the intent behind the flag.